nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -17,34 +17,68 @@
 
 #include <atomic>
 #include <condition_variable>
+#include <memory>
 #include <string>
 #include <thread>
-#include <memory>
 
+#include "lifecycle_msgs/msg/state.hpp"
+#include "nav2_lifecycle/lifecycle_node.hpp"
+#include "nav2_tasks/task_status.hpp"
 #include "rclcpp/rclcpp.hpp"
 #include "std_msgs/msg/empty.hpp"
-#include "nav2_tasks/task_status.hpp"
 
 namespace nav2_tasks
 {
 
 constexpr std::chrono::milliseconds defaultServerTimeout = std::chrono::milliseconds(5000);
 
-template<class CommandMsg, class ResultMsg>
+template<typename CommandMsg, typename ResultMsg>
 const char * getTaskName();
 
-template<class CommandMsg, class ResultMsg>
-class TaskClient
+template<typename CommandMsg, typename ResultMsg>
+class TaskClient : public nav2_lifecycle::LifecycleHelperInterface
 {
 public:
-  explicit TaskClient(rclcpp::Node::SharedPtr & node)
-  : node_(node)
+  explicit TaskClient(nav2_lifecycle::LifecycleNode::SharedPtr node, bool autoinit = false)
+  : node_(node), autoinit_(autoinit)
   {
     resultReceived_ = false;
     statusReceived_ = false;
 
     statusMsg_ = std::make_shared<StatusMsg>();
 
+    // There are some cases where the TaskClient is used from a context that doesn't have
+    // a lifecycle-style inteface (such as BehaviorTrees). For those situations, the
+    // TaskClient class can be automatically configured and activated
+
+    if (autoinit_) {
+      rclcpp_lifecycle::State state0(
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE, "unconfigured");
+      on_configure(state0);
+
+      rclcpp_lifecycle::State state1(
+        lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, "inactive");
```

Comment:
Inactive or configured?

Reply:
Fortunately, the task_client and task_server are now obsolete and have been removed. No longer an issue with the action client since it doesn't have to be activated (at least so far, as it is not lifecycle aware yet).